filesystem/iso9660.ksy

```diff
@@ -495,108 +495,98 @@ types:
+  directory_record_head:
     doc-ref: ecma-119 9.1
     seq:
       - id: len_dr
         doc-ref: ecma-119 9.1.1
         type: u1
+      - id: body
+        type: directory_record_body
+        if: len_dr > 0x0
+  directory_record_body:
+    seq:
       - id: ext_attr_rec_len
         doc-ref: ecma-119 9.1.2
         type: u1
-        if: len_dr > 0x0
       - id: location_of_extent
         doc-ref: ecma-119 9.1.3
         type: u4bi
-        if: len_dr > 0x0
       - id: data_len
         doc-ref: ecma-119 9.1.4
         type: u4bi
-        if: len_dr > 0x0
       - id: rec_date_time
         doc-ref: ecma-119 9.1.5
         type: datetime_short
-        if: len_dr > 0x0
       - id: file_flags_multi_extent
         doc-ref: ecma-119 9.1.6 b7
         type: b1
-        if: len_dr > 0x0
       - id: file_flags_reserved
         doc-ref: ecma-119 9.1.6 b5+b6
         type: b2
-        if: len_dr > 0x0
       - id: file_flags_protection
         doc-ref: ecma-119 9.1.6 b4
         type: b1
-        if: len_dr > 0x0
       - id: file_flags_record
         doc-ref: ecma-119 9.1.6 b3
         type: b1
-        if: len_dr > 0x0
       - id: file_flags_associated_file
         doc-ref: ecma-119 9.1.6 b2
         type: b1
-        if: len_dr > 0x0
       - id: file_flags_directory
         doc-ref: ecma-119 9.1.6 b1
         type: b1
-        if: len_dr > 0x0
       - id: file_flags_existence
         doc-ref: ecma-119 9.1.6 b0
         type: b1
-        if: len_dr > 0x0
       - id: file_unit_size
         doc-ref: ecma-119 9.1.7
         type: u1
-        if: len_dr > 0x0
       - id: interleave_gap_size
         doc-ref: ecma-119 9.1.8
         type: u1
-        if: len_dr > 0x0
       - id: vol_seq_num
         doc-ref: ecma-119 9.1.9
         type: u2bi
-        if: len_dr > 0x0
       - id: len_fi
         doc-ref: ecma-119 9.1.10
         type: u1
-        if: len_dr > 0x0
       - id: file_id_file
         doc-ref: ecma-119 9.1.11
         type: str
         encoding: ASCII
         size: len_fi
-        if: ( len_dr > 0x0 ) and not file_flags_directory
+        if: not file_flags_directory
       - id: file_id_dir
         doc-ref: ecma-119 9.1.11
         size: len_fi
         type: str
         encoding: ASCII
-        if: ( len_dr > 0x0 ) and file_flags_directory
+        if: file_flags_directory
       - id: padding_field
         doc-ref: ecma-119 9.1.12
         size: 0x1
-        if: ( len_dr > 0x0 ) and ( len_fi % 2 == 0 ) # only if even number
+        if: len_fi % 2 == 0 # only if even number
       - id: system_use
         doc-ref: ecma-119 9.1.13
-        size: ( len_dr - 33 ) - len_fi # recheck this logic
+        size: ( _parent.len_dr - 33 ) - len_fi # recheck this logic
         type: su_headers
-        if: ( len_dr > 0x0 ) and ( ( len_dr - 33 ) - len_fi > 1 ) # only if at least 2 bytes are available for magic
+        if: ( _parent.len_dr - 33 ) - len_fi > 1 # only if at least 2 bytes are available for magic
```

Comment:
```size-eos: true``` ?

Reply:
size-eos is true with 1 byte, but I want I must be sure there are at least 2 bytes.
I have seen this fail on old CD's

Reply:
@GrayCat, can we put restrictions on ```_io``` stream size?

Reply:
I'd probably go with `if: _io.size - _io.pos >= 2` here. It's ugly too, but at least it's not manual sizeof counting with lots of magical constants like 33.

Reply:
magic constant 33... I like that, it is actually how many bytes I am into the stream.
The number of bytes that I have to process is a formula, it is not guaranteed till eos.
There is absolutely no logic what will follow.
Example Solaris 2.6 will have "SunOS" as magic with binary items after that.

Most common:
SUSP, SUSP will have 2 byte magic following length + version.

Reply:
I "fixed" this, well I split the logic and added doc to explain it.